lienzo-core/src/main/java/com/ait/lienzo/client/core/image/filter/AbstractImageDataFilter.java

```diff
@@ -0,0 +1,155 @@
+/*
+   Copyright (c) 2017 Ahome' Innovation Technologies. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.image.filter;
+
+import com.ait.lienzo.client.core.Attribute;
+import com.ait.lienzo.client.core.shape.json.AbstractFactory;
+import com.ait.lienzo.client.core.shape.json.validators.ValidationContext;
+import com.ait.lienzo.client.core.shape.json.validators.ValidationException;
+import com.ait.lienzo.shared.core.types.ImageFilterType;
+import com.ait.lienzo.tools.client.collection.MetaData;
+import jsinterop.annotations.JsProperty;
+
+public abstract class AbstractImageDataFilter<T extends AbstractImageDataFilter<T>> implements ImageDataFilter<T> {
+
+    private final MetaData m_meta;
+
+    private final ImageFilterType m_type;
+
+    @JsProperty
+    private boolean active = true;
+
+    protected AbstractImageDataFilter(final ImageFilterType type) {
+        m_type = type;
+
+        m_meta = new MetaData();
+
+        setActive(true);
+    }
+
+    protected AbstractImageDataFilter(final ImageFilterType type, final Object node, final ValidationContext ctx) throws ValidationException {
+        m_type = type;
+
+        if (null == node) {
+            // m_attr = new Attributes(this);
+
+            m_meta = new MetaData();
+
+            return;
+        }
+
+        // @FIXME serialization (mdp)
```

Comment:
Is it still valid? Can we remove it here and below?

Reply:
done